src/Cli/Microsoft.DotNet.Cli.Utils/PathUtility.cs

```diff
@@ -112,9 +112,13 @@ public static bool DeleteFileAndEmptyParents(string path)
             File.Delete(path);
             var dir = Path.GetDirectoryName(path);
 
-            while (!Directory.EnumerateFileSystemEntries(dir).Any())
+            int directoriesDeleted = 0;
+
+            while (!Directory.EnumerateFileSystemEntries(dir).Any() &&
+                directoriesDeleted < maxDirectoriesToDelete)
             {
                 Directory.Delete(dir);
+                directoriesDeleted++;
```

Comment:
Is there a concern that we'll get stuck in this loop?

Reply:
Not really—we'll start out a finite number of directories deep, and that provides an upper limit to how many times it'll run.

Reply:
I wanted to use this as part of cleaning up the workload install records, but I didn't want it to delete directories further than a certain level.  It probably would have been OK to delete empty directories and they would have been recreated if needed, but it felt safer to me to limit it.